map.go

```diff
@@ -55,6 +55,31 @@ type Map struct {
 	fullValueSize int
 }
 
+// NewMapFromFD creates a map from a raw fd.
+//
+// name must be a valid BPF object name.
+//
+// You should not use fd after calling this function.
```

Comment:
Question for semantics: Which of the following would you expect when writing against this interface?

```
fd = MyCodeThatOpensAnFdForMap(...);
map, err := bpf.NewMapFromFD(fd, ...)
if err != nil {
    syscall.Close(fd)
    log("Failed to open map: %s", err)
}
```
Or
```
fd = MyCodeThatOpensAnFdForMap(...);
map, err := bpf.NewMapFromFD(fd, ...)
if err != nil {
    // Fd was already closed by the NewMapFromFD() function
    log("Failed to open map: %s", err)
}
```

I would expect that the FD is still "ours" in the error case, and the fd is now "stolen" in the success case. Particularly if there ends up being any other logic in between the fd open (`defer()`s, function layers, etc).

Reply:
Fixed the code to follow the first style.